src/utils/queue.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const debug = require('debug')('fastboot:ember-app');
+
+/**
+ * Utility Queue class to store queue of sandboxes that can be leveraged when using `buildSandboxPerVisit`.
+ *
+ * @public
+ */
+class Queue {
+  constructor(maxSize = 1) {
+    this.items = [];
+    this.maxSize = maxSize;
+  }
+
+  enqueue(item) {
+    if (this.size() < this.maxSize) {
+      this.items.push(item);
+    } else {
+      debug('Ignoring adding appInstance to queue as Queue is already full!');
```

Comment:
Why silently ignore? Shouldn't this throw an error or return some value indicating that you're not getting what you want?

Or you should document that you must check `isFull()` before calling this to ensure that there's space. Otherwise, I don't know what to expect.

Reply:
The reason to silently ignore is and not throw an error is because when the number of requests is more than the size of the queue, we anyways process the request by creating a sandbox on demand. Throwing an error will block us from doing that. 

Sure I'll document calling `isFull()`.